src/test/tests/applicabilityScan.test.ts

```diff
@@ -0,0 +1,328 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { assert } from 'chai';
+
+import { ConnectionManager } from '../../main/connect/connectionManager';
+import { LogManager } from '../../main/log/logManager';
+import {
+    ApplicabilityRunner,
+    CveApplicableDetails,
+    ApplicabilityScanArgs,
+    ApplicabilityScanResponse
+} from '../../main/scanLogic/scanRunners/applicabilityScan';
+import { ScanUtils } from '../../main/utils/scanUtils';
+import { AnalyzerScanResponse, AnalyzerScanRun, FileIssues, FileRegion } from '../../main/scanLogic/scanRunners/analyzerModels';
+import { IssuesRootTreeNode } from '../../main/treeDataProviders/issuesTree/issuesRootTreeNode';
+import { createDummyCveIssue, createDummyDependencyIssues, createRootTestNode } from './utils/treeNodeUtils.test';
+import { DescriptorTreeNode } from '../../main/treeDataProviders/issuesTree/descriptorTree/descriptorTreeNode';
+import { PackageType } from '../../main/types/projectType';
+import { DependencyScanResults } from '../../main/types/workspaceIssuesDetails';
+import { AnalyzerUtils } from '../../main/treeDataProviders/utils/analyzerUtils';
+import { DependencyIssuesTreeNode } from '../../main/treeDataProviders/issuesTree/descriptorTree/dependencyIssuesTreeNode';
+import { Severity } from '../../main/types/severity';
+import { CveTreeNode } from '../../main/treeDataProviders/issuesTree/descriptorTree/cveTreeNode';
+import { FileTreeNode } from '../../main/treeDataProviders/issuesTree/fileTreeNode';
+import { CodeFileTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/codeFileTreeNode';
+import { IEvidence } from 'jfrog-ide-webview';
+import { CodeIssueTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/codeIssueTreeNode';
+import { ApplicableTreeNode } from '../../main/treeDataProviders/issuesTree/codeFileTree/applicableTreeNode';
+import { removeWindowsWhiteSpace } from './utils/utils.test';
+
+let logManager: LogManager = new LogManager().activate();
+
+describe('Contextual Analysis Scan Tests', () => {
+    const scanApplicable: string = path.join(__dirname, '..', 'resources', 'applicableScan');
+    let tempFolder: string = ScanUtils.createTmpDir();
+
+    [
+        {
+            name: 'One root',
+            file: path.join(scanApplicable, 'requestOneRoot.yaml'),
+            roots: ['/path/to/root'],
+            cves: ['CVE-2021-3918'],
+            skip: []
+        },
+        {
+            name: 'Multiple roots',
+            file: path.join(scanApplicable, 'requestMultipleRoots.yaml'),
+            roots: ['/path/to/root', '/path/to/other'],
+            cves: ['CVE-2021-3918', 'CVE-2021-3807', 'CVE-2022-25878'],
+            skip: ['/path/to/skip']
+        }
+    ].forEach(test => {
+        it('Check generated Yaml request - ' + test.name, () => {
```

Comment:
Check generated Yaml request for one project

Reply:
this is table test, I'm checking here both for one or multiple roots.
It cover both, the name of the test will be with prefix that describes the actual test, after '-' describes the input to test one/multiple projects